packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -139,21 +138,19 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
           return;
         }
 
-        const updatedMaterialsPromise =
-            threeGLTF.correlatedSceneGraph.loadVariant(
-                variantName!, this[$getOnUpdateMethod]);
+        const updatedMaterials =
+            await threeGLTF.correlatedSceneGraph.loadVariant(variantName!);
         const {gltf, gltfElementMap} = threeGLTF.correlatedSceneGraph;
 
-        updatedMaterialsPromise.then(updatedMaterials => {
-          for (const index of updatedMaterials) {
-            const material = gltf.materials![index];
-            this[$model]!.materials[index] = new Material(
-                this[$getOnUpdateMethod],
-                gltf,
-                material,
-                gltfElementMap.get(material) as Set<MeshStandardMaterial>);
-          }
-        });
+        for (const index of updatedMaterials) {
+          const material = gltf.materials![index];
+          this[$model]!.materials[index] = new Material(
+              this[$getOnUpdateMethod](),
+              gltf,
+              material,
+              gltfElementMap.get(material) as Set<MeshStandardMaterial>);
+        }
+        this[$needsRender]();
```

Comment:
I made this PR to fix a bug, and this line was the actual fix. The render was not being updated after a variant was loaded.